tests/runtest.sh

```diff
@@ -84,31 +84,52 @@ run_test_dir()
 restore_coreclr_tests()
 {
     CoreRT_Test_Download_Semaphore=${CoreRT_TestExtRepo}/init-tests.completed
+    CoreRT_NativeArtifact_Download_Semaphore=${CoreRT_TestExtRepo}/init-native-artifact.completed
 
-    if [ -e ${CoreRT_Test_Download_Semaphore} ]; then
+    if [ -e ${CoreRT_Test_Download_Semaphore} ] && [ -e ${CoreRT_NativeArtifact_Download_Semaphore} ]; then
         echo "Tests are already initialized."
         return 0
     fi
 
-    if [ -d ${CoreRT_TestExtRepo} ]; then
-        rm -r ${CoreRT_TestExtRepo}
+    if [ ! -e ${CoreRT_Test_Download_Semaphore} ]; then
```

Comment:
Nit: This could be broken out in a function that restores a URL to a folder we call for each zip file.